src/RawHtmlFieldInput.js

```diff
@@ -0,0 +1,121 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+import {bindActionCreators} from 'redux';
+import {connect} from 'react-redux';
+import {changeBlockValue} from './actions';
+import {
+  getFieldName,
+  isStatic,
+  shouldRunInnerScripts
+} from './processing/utils';
+
+
+const MutationObserver = window.MutationObserver
+                      || window.WebKitMutationObserver
+                      || window.MozMutationObserver;
+
+
+@connect(null, (dispatch, props) => {
+  const {fieldId, blockId} = props;
+  return bindActionCreators({
+    changeBlockValue: value => changeBlockValue(fieldId, blockId, value),
+  }, dispatch);
+})
+class RawHtmlFieldInput extends React.Component {
+  static propTypes = {
+    fieldId: PropTypes.string.isRequired,
+    blockDefinition: PropTypes.object.isRequired,
+    blockId: PropTypes.string.isRequired,
+    html: PropTypes.string.isRequired,
+    value: PropTypes.any,
+  };
+
+  runInnerScripts() {
+    if (shouldRunInnerScripts(this.props.blockDefinition)) {
+      for (let script
+           of ReactDOM.findDOMNode(this).querySelectorAll('script')) {
+        script.parentNode.removeChild(script);
+        window.eval(script.innerHTML);
+      }
+    }
+  }
+
+  setValue(input) {
+    const {value} = this.props;
+    if ((value !== undefined) && (value !== null)) {
+      if (input.type === 'file') {
+        input.files = value;
+      } else if ((input.type === 'checkbox') || (input.type === 'radio')) {
+        input.checked = value === null ? false : (
+          typeof value === 'boolean' ? value : value.includes(input.value));
+      } else if (input.type === 'hidden') {
+        input.value = value;
+        input.dispatchEvent(new Event('change'));
```

Comment:
Why is this only done for hidden inputs?

Reply:
Because changing the value on an hidden input does not trigger a change event. I discovered that the hard way :(
As you can see here, no event is triggered by hidden inputs:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Input/hidden